server/node/src/main/java/org/kaaproject/kaa/server/operations/service/cache/concurrent/ConcurrentCacheService.java

```diff
@@ -323,7 +323,7 @@ public String putConfId(ConfigurationIdKey key, String value) {
                         relatedChanges.add(historyDto);
                     } else if (changeType == ChangeType.ADD_PROF || changeType == ChangeType.REMOVE_PROF) {
                         ProfileFilterDto profileFilter = profileService.findProfileFilterById(changeDto.getProfileFilterId());
-                        if (supports(profileFilter, key.getEndpointProfileSchemaVersion(), key.getServerProfileSchemaVersion())) {
+                        if (profileFilter != null && supports(profileFilter, key.getEndpointProfileSchemaVersion(), key.getServerProfileSchemaVersion())) {
```

Comment:
You can either introduce "deleted" flag for profile filters or copy this data to other table. I would use "deleted" flag.